lib/rack/utils.rb

```diff
@@ -374,14 +374,22 @@ def get_byte_ranges(http_range, size)
     # that have already been processed by HMAC. This should not be used
     # on variable length plaintext strings because it could leak length info
     # via timing attacks.
-    def secure_compare(a, b)
-      return false unless a.bytesize == b.bytesize
+    if defined?(OpenSSL.secure_compare)
```

Comment:
It looks odd that we are checking for `OpenSSL.secure_compare` but using `OpenSSL.fixed_length_secure_compare`.

Reply:
Woops got my wires crossed there - fixed! Thanks.